docs/src/reference/gremlin-variants.asciidoc

```diff
@@ -1897,3 +1897,300 @@ pip install gremlinpython
 pip install aiohttp
 python example.py
 ----
+
+[[gremlin-go]]
+== Gremlin-Go
+
+image:gremlin-go.png[width=130,float=right] Apache TinkerPop's Gremlin-Go implements Gremlin within the link:https://go.dev/[Go] language and can therefore be used on different operating systems. Go's syntax has the similar constructs as Java including
+"dot notation" for function chaining (`a.b.c`) and round bracket function arguments (`a(b,c)`). Something unlike Java is that Gremlin-Go requires a 
+`gremlingo` prefix when using the namespace (`a(b())` vs `gremlingo.a(gremlingo.__.b())`). Anyone familiar with Gremlin-Java will be able to work
+with Gremlin-Go with relative ease. Moreover, there are a few added constructs to Gremlin-Go that make traversals a bit more
+succinct.
+
+To install the Gremlin-Go as a dependency for your project, run the following in the root directory of your project that contains your `go.mod` file:
+
+[source,bash]
+----
+go get github.com/apache/tinkerpop/gremlin-go/v3[optionally append @<version>, such as @v3.5.3]
+----
+
+[[gremlin-go-connecting]]
+=== Connecting
+
+The pattern for connecting is described in <<connecting-gremlin,Connecting Gremlin>> and it basically distills down to
+creating a `GraphTraversalSource`. A `GraphTraversalSource` is created from the anonymous `Traversal_()`.
+
+[source,go]
+----
+remote, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+g := gremlingo.Traversal_().WithRemote(remote)
+----
+
+If you need to additional parameters to connection setup, you can pass in a configuration function.
+
+[source,go]
+----
+remote, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
+  func(settings *DriverRemoteConnectionSettings) {
+    settings.TraversalSource = "gmodern"
+  })
+----
+
+Gremlin-go supports plain text authentication. It can be set in the connection function.
+
+[source,go]
+----
+remote, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
+  func(settings *DriverRemoteConnectionSettings) {
+    settings.TlsConfig = &tls.Config{InsecureSkipVerify: true}
+    settings.AuthInfo = gremlingo.BasicAuthInfo("login", "password")
+  })
+----
+
+If you authenticate to a remote <<connecting-gremlin-server,Gremlin Server>> or
+<<connecting-rgp,Remote Gremlin Provider>>, this server normally has SSL activated and the websockets url will start
+with 'wss://'.
+
+Some connection options can also be set on individual requests made through the using `With()` step on the
+`TraversalSource`. For instance to set request timeout to 500 milliseconds:
+
+[source,go]
+----
+results, err := g.With("evaluationTimeout", 500).V().Out("knows").ToList()
+----
+
+The following options are allowed on a per-request basis in this fashion: `batchSize`, `requestId`, `userAgent` and
+`evaluationTimeout` (formerly `scriptEvaluationTimeout` which is also supported but now deprecated).
+
+anchor:go-imports[]
+[[gremlin-go-imports]]
+=== Common Imports
```

Comment:
this section should discuss the pattern of aliasing (i.e. assigning to `var`/`const`) to get a more Gremlin like feel:

```go
const Local = gremlingo.Local
var Neq = gremlingo.P.Neq
var __ = gremlingo.T__
var In = __.In
```

On the subject of naming, I think `T__` has been controversial for a while now but no one has come up with anything groundbreakingly better. The lesser of two evils is to just go with renaming `T__` to the full naming of `AnonymousTraversal` and then using the aliasing above to get it to be `__` like regular Gremlin. Furthermore, I think that enums like `Local` probably need their associated prefixes as in `Scope.Local`.

Reply:
Both of these things are going to be worked on separately from this ticket, thanks for raising this!